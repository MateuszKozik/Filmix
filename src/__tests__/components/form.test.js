import React from "react";
import { render } from "@testing-library/react";
import { Form } from "../../components";
import { Link } from "react-router-dom";
jest.mock("react-router-dom");

describe("<Form />", () => {
	it("renders the <Form /> with populated data", () => {
		const { container, getByText, getByPlaceholderText } = render(
			<Form>
				<Form.Title>Sign In Now</Form.Title>

				<Form.Base>
					<Form.Input placeholder="Email address" onChange={() => {}} />
					<Form.Input
						type="password"
						placeholder="Password"
						onChange={() => {}}
					/>
					<Form.Submit type="submit" disabled>
						Sign In
					</Form.Submit>
				</Form.Base>

				<Form.Text>New to Filmix? Sign up now.</Form.Text>
				<Form.TextSmall>
					Register to watch Your favourite movies!
				</Form.TextSmall>
			</Form>
		);

		expect(getByText("Register to watch Your favourite movies!")).toBeTruthy();
		expect(getByText("Sign In")).toBeTruthy();
		expect(getByText("Sign In").disabled).toBeTruthy();
		expect(getByPlaceholderText("Email address")).toBeTruthy();
		expect(getByPlaceholderText("Password")).toBeTruthy();
		expect(container.firstChild).toMatchSnapshot();
	});

	it("rendest the <Form /> with an error", () => {
		const { container, getByText, queryByText } = render(
			<Form>
				<Form.Error>Passowrd is incorrect</Form.Error>
				<Form.Submit type="submit">Sign In</Form.Submit>
			</Form>
		);

		expect(getByText("Passowrd is incorrect")).toBeTruthy();
		expect(queryByText("Sign In").disabled).toBeFalsy();
		expect(container.firstChild).toMatchSnapshot();
	});
});
