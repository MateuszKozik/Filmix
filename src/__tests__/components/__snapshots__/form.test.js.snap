// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders the <Form /> with populated data 1`] = `
<div
  class="form__Container-sc-1bwtepx-0 jkVpBN"
>
  <h1
    class="form__Title-sc-1bwtepx-3 bbPhpT"
  >
    Sign In Now
  </h1>
  <form
    class="form__Base-sc-1bwtepx-2 elsYFw"
  >
    <input
      class="form__Input-sc-1bwtepx-7 fZMeHz"
      placeholder="Email address"
    />
    <input
      class="form__Input-sc-1bwtepx-7 fZMeHz"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Submit-sc-1bwtepx-8 bcgRDK"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="form__Text-sc-1bwtepx-4 vILRm"
  >
    New to Filmix? Sign up now.
  </p>
  <p
    class="form__TextSmall-sc-1bwtepx-5 iSEJhm"
  >
    Register to watch Your favourite movies!
  </p>
</div>
`;

exports[`<Form /> rendest the <Form /> with an error 1`] = `
<div
  class="form__Container-sc-1bwtepx-0 jkVpBN"
>
  <div
    class="form__Error-sc-1bwtepx-1 ciJKBy"
  >
    Passowrd is incorrect
  </div>
  <button
    class="form__Submit-sc-1bwtepx-8 bcgRDK"
    type="submit"
  >
    Sign In
  </button>
</div>
`;
